src/rust/ensogl/src/control/callback.rs

```diff
@@ -132,9 +132,8 @@ impl<T> Registry1<T> {
 // === CopyRegistry ===
 // ====================
 
-/// Registry gathering callbacks. Each registered callback is assigned with a handle. Callback and
-/// handle lifetimes are strictly connected. As soon a handle is dropped, the callback is removed
-/// as well.
+/// Specialized version of `Registry` for arguments implementing `Copy`. Passing copy-able elements
+/// as values is more performant than by reference.
```

Comment:
Did you benchmarked it? I would say, that rust optimizer will handle such case anyway (because it know, that is "readonly refrence" and cannot be changed by other process, so it can replace reference with copy!